File: src/main/java/no/nav/bidrag/beregn/forskudd/core/beregning/ForskuddBeregning.java
package no.nav.bidrag.beregn.forskudd.core.beregning;

import no.nav.bidrag.beregn.forskudd.core.bo.GrunnlagBeregning;
import no.nav.bidrag.beregn.forskudd.core.bo.ResultatBeregning;

public interface ForskuddBeregning {

  ResultatBeregning beregn(GrunnlagBeregning grunnlag);

  static ForskuddBeregning getInstance() {
    return new ForskuddBeregningImpl();
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/forskudd/core/beregning/ForskuddBeregning.kt
package no.nav.bidrag.beregn.forskudd.core.beregning

import no.nav.bidrag.beregn.forskudd.core.bo.GrunnlagBeregning
import no.nav.bidrag.beregn.forskudd.core.bo.ResultatBeregning

fun interface ForskuddBeregning {
    fun beregn(grunnlag: GrunnlagBeregning): ResultatBeregning

    companion object {
        fun getInstance(): ForskuddBeregning {
            return ForskuddBeregningImpl()
        }
    }
}
